### PR TITLE
anti corruption layer - Issue 1325.

### LINKED ISSUE #1325: Anti-Corruption Layer pattern
https://docs.microsoft.com/en-us/azure/architecture/patterns/anti-corruption-layer

### PR DESCRIPTION
**Pull request by besok**
closes #1325 

Describes the anti-corruption pattern
- added a new pattern

### CODE REVIEW

**Comment by besok** [comment]
do i need to fix sonar issues? 
The tests here just to demonstrate how the pattern works.

**Comment by besok** [comment]
Hello @iluwatar!
Could you please give a look at this PR?

File: anti-corruption-layer/src/main/java/com/iluwatar/corruption/system/legacy/LegacyOrder.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
+package com.iluwatar.corruption.system.legacy;
+
+import java.util.Objects;
+
+/**
+ * The class represents an order in the legacy system.
+ * The class is used by the legacy system to store the data.
+ */
+public class LegacyOrder {
```
Comment (iluwatar) [inline anti-corruption-layer/src/main/java/com/iluwatar/corruption/system/legacy/LegacyOrder.java:9]: Use Lombok to get rid of boilerplate such as getters, setters, constructors, etc.

File: anti-corruption-layer/src/main/java/com/iluwatar/corruption/system/modern/Customer.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package com.iluwatar.corruption.system.modern;
+
+import java.util.Objects;
+
+/**
+ * The class represents a customer in the modern system.
+ */
+public class Customer {
```
Comment (iluwatar) [inline anti-corruption-layer/src/main/java/com/iluwatar/corruption/system/modern/Customer.java:8]: Use Lombok to get rid of boilerplate such as getters, setters, constructors, equals, hashCode, etc.

File: anti-corruption-layer/src/main/java/com/iluwatar/corruption/system/modern/ModernOrder.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  private String extra;
+
+  /**
+   * Constructor.
+   * @param id the id of the order
+   * @param customer the customer of the order
+   * @param shipment the shipment of the order
+   * @param extra  the extra data of the order
+   */
+  public ModernOrder(String id, Customer customer, Shipment shipment, String extra) {
```
Comment (iluwatar) [inline anti-corruption-layer/src/main/java/com/iluwatar/corruption/system/modern/ModernOrder.java:23]: Use Lombok to get rid of boilerplate such as getters, setters, constructors, etc.

File: anti-corruption-layer/src/main/java/com/iluwatar/corruption/system/modern/Shipment.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
+package com.iluwatar.corruption.system.modern;
+
+import java.util.Objects;
+
+/**
+ * The class represents a shipment in the modern system.
+ * The class is used by the modern system to store the data.
+ */
+public class Shipment {
```
Comment (iluwatar) [inline anti-corruption-layer/src/main/java/com/iluwatar/corruption/system/modern/Shipment.java:9]: Use Lombok to get rid of boilerplate such as getters, setters, constructors, etc.

File: anti-corruption-layer/src/main/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1 @@
+spring.main.allow-circular-references=true
```
Comment (iluwatar) [inline anti-corruption-layer/src/main/resources/application.properties:1]: Hmm, looks smelly. Any way to get rid of circular references?

**Review by iluwatar — CHANGES_REQUESTED** [review]

File: anti-corruption-layer/src/main/resources/application.properties (hunk 5 above)
Comment (besok) [reply anti-corruption-layer/src/main/resources/application.properties]: yeah, I added two-way links and apparently, we can keep only one way when we check the existence of the order in the old shop from the modern one.

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Looks good! Thank you for the contribution 🎉

@all-contributors please add @besok for code

**Pull request merged**